CMakeLists.txt

```diff
@@ -104,10 +104,25 @@ if (HYDRO_EB)
    add_subdirectory(Redistribution)
 endif ()
 
+function (set_cpp_sources_to_cuda_arch _target)
+   get_target_property(_sources ${_target} SOURCES)
+   list(FILTER _sources INCLUDE REGEX "\\.cpp$")
+   set_source_files_properties(${_sources} PROPERTIES CUDA_ARCHITECTURES "${AMREX_CUDA_ARCHS}")
+endfunction ()
+
 if (HYDRO_GPU_BACKEND STREQUAL "CUDA")
    enable_language(CUDA)
    #setup_target_for_cuda_compilation(amrex_hydro)
+
```

Comment:
Hey @ax3l, Can you take a look at this hacky workaround? This doesn't feel like the way. -- Thanks!